### ISSUE TITLE
"Ghost" monitor (Cinnamon beta)

### ISSUE THREAD

**Issue by MiZoG99** [linuxmint/mint22-beta#31]
Mint 22 Cinnamon beta: A "ghost monitor" appears in my system.

```
  Monitor-1: HDMI-A-1 mapped: HDMI-0 note: disabled pos: primary
    model: AOC 24E1W1 serial: PPCP1HA001165 built: 2023 res: 1920x1080 hz: 60
    dpi: 93 gamma: 1.2 size: 527x296mm (20.75x11.65") diag: 604mm (23.8")
    ratio: 16:9 modes: max: 1920x1080 min: 640x480
  Monitor-2: Unknown-1 mapped: None-1-1 note: disabled size-res: N/A
    modes: 1024x768

```
[Full system specs (inxi)](https://termbin.com/s7t2)

xrandr output
```
Screen 0: minimum 8 x 8, current 1920 x 1080, maximum 16384 x 16384
VGA-0 disconnected (normal left inverted right x axis y axis)
DVI-D-0 disconnected (normal left inverted right x axis y axis)
HDMI-0 connected primary 1920x1080+0+0 (normal left inverted right x axis y axis) 527mm x 296mm
   1920x1080     60.00*+  59.94    50.00    60.00    50.04  
   1680x1050     59.95  
   1440x900      59.89  
   1280x1024     75.02    60.02  
   1280x960      60.00  
   1280x720      60.00    59.94    50.00  
   1024x768      75.03    70.07    60.00  
   800x600       75.00    72.19    60.32    56.25  
   720x576       50.00  
   720x480       59.94  
   640x480       75.00    72.81    59.94    59.93  
None-1-1 connected (normal left inverted right x axis y axis)
   1024x768      60.00 +


```
Integrated intel graphics is disabled in UEFI/Bios settings.

"Real" monitor in Display Settings

![RealMonitor](https://github.com/linuxmint/mint22-beta/assets/137620210/43e81028-74c7-4673-96a2-b7000470fbdf)

Ghost monitor in Display Settings

![BogusMonitor](https://github.com/linuxmint/mint22-beta/assets/137620210/41bf4c9d-ce1c-452e-96f4-72f36a7f37e0)

Slight buggy behaviour results from this. A screenshot taken assumes my screen size comprises both the real and imaginary monitors:
![FakeScreenResolution](https://github.com/linuxmint/mint22-beta/assets/137620210/17f694fe-a6ee-4903-bebd-7267d5cc8b55)

Occurrences of screen tearing not present in previous Cinnamon builds are perhaps related to this bug. This is just an assumption though.

Most interestingly this "ghost" monitor is not present upstream.
Current Ubuntu 24.04, same base system, same hardware, same nvidia driver version, same X server:

![Upstream](https://github.com/linuxmint/mint22-beta/assets/137620210/db8e05d0-6e35-409a-9958-bb82f5fb9d8a)

Suffice to say this was neither present in Cinnamon 6.0.4 in both Mint 21.3 / LMDE 6

**Comment by JosephMcc** [comment]
Did you happen to restart Cinnamon before this showed up? Doesn't happen to me on first boot, only after a Cinnamon restart.

**Comment by MiZoG99** [comment]
I didn't notice it early to say when it did appear for a first time.

**Comment by Harry-W-Haines-III** [comment]
I noticed the same thing myself but I'm not sure its related to Cinnamon.  It looks like it has something to do with the NVIDIA driver and kernel.  Take a look at this:

nvidia with nouveau driver:

inxi -Gxxx
Graphics:
  Device-1: NVIDIA GA107 [GeForce RTX 3050 6GB] vendor: Micro-Star MSI
    driver: nouveau v: kernel arch: Ampere ports: active: DP-1
    empty: HDMI-A-1,HDMI-A-2 bus-ID: 0000:01:00.0 chip-ID: 10de:2584
    class-ID: 0300
  Display: x11 server: X.Org v: 21.1.11 with: Xwayland v: 23.2.6 driver: X:
    loaded: modesetting unloaded: fbdev,vesa dri: nouveau gpu: nouveau
    display-ID: :0 screens: 1
  Screen-1: 0 s-res: 2560x1440 s-dpi: 96 s-size: 677x381mm (26.65x15.00")
    s-diag: 777mm (30.58")
  Monitor-1: DP-1 model: HP OMEN 32q serial: CNC4140L69 res: 2560x1440
    hz: 60 dpi: 93 size: 697x393mm (27.44x15.47") diag: 800mm (31.5") modes:
    max: 2560x1440 min: 720x400
  API: EGL v: 1.5 hw: drv: nvidia nouveau platforms: device: 0 drv: nouveau
    device: 1 drv: swrast gbm: drv: nouveau surfaceless: drv: nouveau x11:
    drv: nouveau inactive: wayland
  API: OpenGL v: 4.5 compat-v: 4.3 vendor: mesa v: 24.0.5-1ubuntu1
    glx-v: 1.4 direct-render: yes renderer: NV177 device-ID: 10de:2584

nvidia  with driver: nvidia v: 535.183.01  (current general release version)

inxi -Gxxx
Graphics:
  Device-1: NVIDIA GA107 [GeForce RTX 3050 6GB] vendor: Micro-Star MSI
    driver: nvidia v: 535.183.01 arch: Ampere ports: active: none off: DP-1
    empty: HDMI-A-1,HDMI-A-2 bus-ID: 0000:01:00.0 chip-ID: 10de:2584
    class-ID: 0300
  Display: x11 server: X.Org v: 21.1.11 with: Xwayland v: 23.2.6 driver: X:
    loaded: modesetting,nouveau,nvidia unloaded: fbdev,vesa
    gpu: nvidia,nvidia-nvswitch display-ID: :0 screens: 1
  Screen-1: 0 s-res: 2560x1440 s-dpi: 96 s-size: 677x381mm (26.65x15.00")
    s-diag: 777mm (30.58")
  Monitor-1: DP-1 mapped: DP-0 note: disabled pos: primary
    model: HP OMEN 32q serial: CNC4140L69 res: 2560x1440 hz: 60 dpi: 93
    size: 697x393mm (27.44x15.47") diag: 800mm (31.5") modes: max: 2560x1440
    min: 640x480
  Monitor-2: Unknown-1 mapped: None-1-1 note: disabled size-res: N/A
    modes: 2560x1440
  API: EGL v: 1.5 hw: drv: nvidia platforms: device: 0 drv: nvidia device: 2
    drv: swrast gbm: drv: kms_swrast surfaceless: drv: nvidia x11: drv: nvidia
    inactive: wayland,device-1
  API: OpenGL v: 4.6.0 compat-v: 4.5 vendor: nvidia mesa v: 535.183.01
    glx-v: 1.4 direct-render: yes renderer: NVIDIA GeForce RTX 3050/PCIe/SSE2

**Comment by Harry-W-Haines-III** [comment]
More info here about this issue.
https://www.reddit.com/r/pop_os/comments/1bvguqd/solvedfixfixed_extra_unknown_display_after_kernel/

**Comment by Harry-W-Haines-III** [comment]
Work around mentioned here:
https://bugs.launchpad.net/ubuntu/+source/linux/+bug/2060268

**Comment by MiZoG99** [comment]
The strange thing is that Ubuntu 24.04 (I can switch back n forth between clonezilla OS images) with the same nvidia driver (535.x) is not affected by this. That actually made me to open this issue. My testing machine has seen dozens of installations and never experienced this bug. I will look into the workarounds soon when I get back to that machine.

**Comment by anaximeno** [comment]
There were some changes on muffin regarding the monitor manager, I wonder if they are related by any means https://github.com/linuxmint/muffin/commits/master/

**Comment by Harry-W-Haines-III** [comment]
I ran this suggested fix on a testing box = same hardware as this one.  It seemed to fix the issue.  https://bugs.launchpad.net/ubuntu/+source/linux/+bug/2060268/comments/32

**Comment by MiZoG99** [comment]
I do confirm that adding
`initcall_blacklist=simpledrm_platform_driver_ini`
as a boot parameter makes ghost monitor go away.

Still there hasn't been any such issue with nvidia-driver on any other distro
and most importantly on Ubuntu 24.04 with the same version of the driver.

**Comment by Harry-W-Haines-III** [comment]
[MiZoG99](https://github.com/MiZoG99) Please read carefully that bug report.
https://bugs.launchpad.net/ubuntu/+source/linux/+bug/2060268

**Comment by Harry-W-Haines-III** [comment]
Just found this same issue in POP!_OS, ISO number 41 (the current one for download).
![Screenshot from 2024-07-02 01-15-20](https://github.com/linuxmint/mint22-beta/assets/45599505/030d66ad-2541-4bee-9bdc-fbdeda08cf4e)

![Screenshot from 2024-07-04 21-22-18](https://github.com/linuxmint/mint22-beta/assets/45599505/57f604b6-dedc-4292-8f88-757479df60db)

**Comment by Harry-W-Haines-III** [comment]
It is odd that it doesn't show up in Ubuntu.  Perhaps they patched the issue?
Here is even more information.
https://github.com/pop-os/pop/issues/3257

**Comment by Harry-W-Haines-III** [comment]
After installing and updating POP!_OS to the 6.9.3 kernel the "Ghost" monitor issue went away.  FYI I also run the Nvidia driver v: 550.90.07 in the LM 22 Beta as well as LM 21.3.  It is in the Ubuntu 22.04 repository.  Can we now safely assume this is an 'upstream' issue?

![Screenshot from 2024-07-05 08-51-42](https://github.com/linuxmint/mint22-beta/assets/45599505/e98f81f3-3313-43fd-bf4d-9ea379a8321a)

**Comment by Harry-W-Haines-III** [comment]
Here with LM 22 Cinnamon fully updated using the xanmod 6.9.7 kernel.

![Screenshot from 2024-07-05 13-07-34](https://github.com/linuxmint/mint22-beta/assets/45599505/a3c76886-0617-4d49-ad4e-9a330ec6ceb5)

**Comment by StercusMax** [comment]
Hello,

This is an upstream issue related to the NVIDIA driver, not a Cinnamon issue. I experienced the same problem on Ubuntu 24.04 (when it was out) and on linux mint 21 (so it's not a really new issue, maybe we didn't have the same driver). If I remember correctly, there are some possible workarounds to remove this screen, with grub settings.

**Comment by StercusMax** [comment]
We can close this issue, i guess

**Comment by clefebvre** [comment]
> The strange thing is that Ubuntu 24.04 (I can switch back n forth between clonezilla OS images) with the same nvidia driver (535.x) is not affected by this. That actually made me to open this issue. My testing machine has seen dozens of installations and never experienced this bug. I will look into the workarounds soon when I get back to that machine.

Are you sure about this? They use GNOME on top Wayland by default afaik.

Did you reproduce it with GNOME+Xorg in Ubuntu 24.04? And how about Xubuntu (Xfce+Xorg) 24.04?

**Comment by StercusMax** [comment]
That's related to xorg and nvidia driver, i tested on ubuntu 24.04 like one week ago with xorg and it had a ghost monitor, another flavour i tested was ubuntu unity and the same issue again, xubuntu should probably get the same issue.

**Comment by StercusMax** [comment]
(nvidia driver 535, 550 are affected, i used theses two different one on ubuntu gnome and ubuntu unity)

**Comment by Harry-W-Haines-III** [comment]
I never saw this in LM 21.3.  Noticed it right away in the 22 Beta with the same 535 drivers.  I highly doubt its an Xorg issue because as soon as I move off the the Ubuntu 6.8.xx kernel it is gone.  I keep seeing a message at the end of the load in dmesg that I never saw before for:

"simple-framebuffer simple-framebuffer.0: swiotlb buffer is full (sz: 1470464 bytes), total 32768 (slots), used 304 (slots)"

But there shouldn't be a simple-framebuffer IF the Nvidia drivers are loading.

Precisely what 'initcall_blacklist=simpledrm_platform_driver_init' in /etc/default/grub solves.

**Comment by StercusMax** [comment]
> I never saw this in LM 21.3. Noticed it right away in the 22 Beta with the same 535 drivers. I highly doubt its an Xorg issue because as soon as I move off the the Ubuntu 6.8.xx kernel it is gone. I keep seeing a message at the end of the load in dmesg that I never saw before for:
> 
> "simple-framebuffer simple-framebuffer.0: swiotlb buffer is full (sz: 1470464 bytes), total 32768 (slots), used 304 (slots)"
> 
> But there shouldn't be a simple-framebuffer IF the Nvidia drivers are loading.
> 
> Precisely what 'initcall_blacklist=simpledrm_platform_driver_init' in /etc/default/grub solves.

Yes you are right, mb, i just found this post about this issue: https://bbs.archlinux.org/viewtopic.php?id=288009

**Comment by StercusMax** [comment]
I however had the issue in linux mint 21, probably because of the edge version

**Comment by Harry-W-Haines-III** [comment]
I just checked, it is still there with the Nvidia 555 drivers and Ubuntu 6.8 kernel.  Its not really a problem for me anymore because I have given up with the poor quality of the Ubuntu kernels and have moved to third party kernels.  This is clearly an upstream issue anyway.

**Comment by StercusMax** [comment]
The only thing where i was right XD, on my side fedora doesn't have the issue, so you are right

**Comment by MiZoG99** [comment]
> > The strange thing is that Ubuntu 24.04 (I can switch back n forth between clonezilla OS images) with the same nvidia driver (535.x) is not affected by this. That actually made me to open this issue. My testing machine has seen dozens of installations and never experienced this bug. I will look into the workarounds soon when I get back to that machine.
> 
> Are you sure about this? They use GNOME on top Wayland by default afaik.
> 
> Did you reproduce it with GNOME+Xorg in Ubuntu 24.04? And how about Xubuntu (Xfce+Xorg) 24.04?

Yes it is x11. Sure. Ubuntu (main plain Ubuntu) defaults to x11 with my nvidia gtx 750 ti.
Same kernel, exact same version of the nvidia driver, both systems on x11. Weird that I don't have the ghost monitor on Ubuntu.

**Comment by MiZoG99** [comment]
I'm closing this issue
xubuntu 24.04 suffers OOB from the very same issue.
 
![ghostmonitoronxubuntuaswell](https://github.com/user-attachments/assets/f4696c19-2e88-4eb5-b10e-8059cec867dd)

Now why this is not happening on main Ubuntu 24.04 I can't explain. Due to something that eludes me right now I guess. Still I have plenty of evidence this is not a Mint-related bug.

**Comment by ghost** [comment]
Noticed this on my end too. However I don't know how, but as soon as I was about to write here, the ghost monitor just literally - and I mean literally - disappeared by its own, without any manual intervention (speaking about Cinnamon)

Restarted my system and now there's no ghost 2nd monitor. 

Thats very odd

**Comment by StercusMax** [comment]
@vajdao The second monitor freaked out XD, weird behavior, i personnally had the second ghost monitor in the past in ubuntu 24.04, the issue is probably with the kernel.

**Comment by ghost** [comment]
@Ikytsu yeah, the most interesting part was when I was looking confused at my machine, because truly I have two monitors, but the second one was powered off... xD so I was like, ok lets unplug all the cables and see what happens... still the ghosted one stood there like a creep. Then it suddenly vanished. Just like a ghost. D'oh 😂
